apps/newsletters-api/src/app/routes/newsletters.ts

```diff
@@ -105,4 +107,51 @@ export function registerReadWriteNewsletterRoutes(app: FastifyInstance) {
 
 		return makeSuccessResponse(storageResponse.data);
 	});
+
+	app.post<{
+		Params: { newsletterId: string };
+		Body: unknown;
+	}>('/api/newsletters/:newsletterId', async (req, res) => {
+		const user = getUserProfile(req); //Todo: create middleware to get user profile
+		const accessDeniedError = await makeAccessDeniedApiResponse(
+			user.profile,
+			'editNewsletters',
+		);
+		if (accessDeniedError) {
+			return res.status(403).send(accessDeniedError);
+		}
+
+		const { newsletterId } = req.params;
+		const { body: newsletter } = req;
+		const newsletterIdAsNumber = Number(newsletterId);
+		const newsletterData = newsletter as NewsletterData;
+
+		if (isNaN(newsletterIdAsNumber)) {
+			return res.status(400).send(makeErrorResponse(`Non numeric id provided`));
+		}
+
+		replaceNullWithUndefinedForUnknown(newsletterData);
+
+		if (!isNewsletterData(newsletterData)) {
+			return res.status(400).send(makeErrorResponse(`Not a valid newsletter`));
+		}
+
```

Comment:
I think it would be better to re-order this so we avoid unsafely casting `newsletter as NewsletterData` (I know its only a few lines before it gets validated) - if we do `replaceNukkWithUndefinedForUnknown`, then validate with `isNewsletterData`, `newsletter` will be type-safe